data_model/src/lib.rs

```diff
@@ -150,8 +129,28 @@ impl Debug for Name {
     }
 }
 
-/// Represents a sequence of bytes. Used for storing encoded data.
-pub type Bytes = Vec<u8>;
+impl<'de> Deserialize<'de> for Name {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: serde::Deserializer<'de>,
+    {
+        #[cfg(not(feature = "std"))]
+        use alloc::borrow::Cow;
+        #[cfg(feature = "std")]
+        use std::borrow::Cow;
+
+        use serde::de::Error as _;
+
+        let name = <Cow<str>>::deserialize(deserializer)?;
+        Name::from_str(&name).map_err(D::Error::custom)
+    }
+}
```

Comment:
I've found [this](https://docs.rs/serde_with/1.3.1/serde_with/rust/display_fromstr/index.html) crate which helps with boilerplate like that when you want to use `from_str()` for deserialization.

Just to keep in mind, if there will be a lot of places like this

Reply:
I already opened #1995 for this